web/src/app/util/Search.js

```diff
@@ -51,21 +51,26 @@ const useStyles = makeStyles((theme) => {
  */
 export default function Search(props) {
   const [searchParam, setSearchParam] = useURLParam('search', '')
+  // track the last value so we know if it changed externally
+  // or from a local event so we don't lose typed characters.
+  const [searchValue, setSearchValue] = useState(searchParam)
```

Comment:
is this a duplicate instantiation of line 59? `const [search, setSearch] = useState(searchParam)`

Reply:
No, but this should have a more clear name as to its purpose. This state variable tracks the previous param value so we can distinguish the difference between the URL changing (e.g. from a filter popover) and updating the URL because the user typed in the search field directly.

Reply:
Basically this component will call `setSearchParam` and `setSearchValue` together, but external changes will only result in `searchParam` changing so that we can use `prevParamValue !== searchParam` to determine an external change.